nexus/tests/integration_tests/volume_management.rs

```diff
@@ -1511,6 +1511,289 @@ async fn test_volume_remove_rop_saga_deleted_volume(
+
+    // Request again, we should see the incremented values now..
+    let new_vol = datastore.volume_get(volume_id).await.unwrap();
+    volume_match_gen(new_vol, vec![0, 8, 10]);
+}
+#[nexus_test]
+async fn test_volume_get_updates_sparse_mid_multiple_gen(
+    cptestctx: &ControlPlaneTestContext,
+) {
+    // Verify that a volume_get will update the generation number in the
+    // database when the volume type is Volume with multiple sub_volumes of
+    // type Region and also verify that a non generation sub_volume in the
+    // middle of the sub_volumes won't be a problem
+    let nexus = &cptestctx.server.apictx.nexus;
+    let datastore = nexus.datastore();
+    let volume_id = Uuid::new_v4();
+    let block_size = 512;
+
+    // Create three sub_vols.
+    let subvol_one = create_region(block_size, 7, Uuid::new_v4());
+    let subvol_two = VolumeConstructionRequest::File {
+        id: Uuid::new_v4(),
+        block_size,
+        path: "/lol".to_string(),
+    };
+    let subvol_three = create_region(block_size, 9, Uuid::new_v4());
+
+    // Make the volume with our three sub_volumes
+    let volume_construction_request = VolumeConstructionRequest::Volume {
+        id: volume_id,
+        block_size,
+        sub_volumes: vec![subvol_one, subvol_two, subvol_three],
+        read_only_parent: None,
+    };
+
+    // Insert the volume into the database.
+    datastore
+        .volume_create(nexus_db_model::Volume::new(
+            volume_id,
+            serde_json::to_string(&volume_construction_request).unwrap(),
+        ))
+        .await
+        .unwrap();
+
+    // The first time back, we get our original values, but internally the
+    // generation number goes up.
+    let new_vol = datastore.volume_get(volume_id).await.unwrap();
+    volume_match_gen(new_vol, vec![7, 0, 9]);
+
+    // Request again, we should see the incremented values now..
+    let new_vol = datastore.volume_get(volume_id).await.unwrap();
+    volume_match_gen(new_vol, vec![8, 0, 10]);
+}
+
+// Test function that creates a VolumeConstructionRequest::Region With gen,
+// and UUID you passed in.
+fn create_region(
+    block_size: u64,
+    gen: u64,
+    id: Uuid,
+) -> VolumeConstructionRequest {
+    VolumeConstructionRequest::Region {
+        block_size,
+        gen,
+        opts: CrucibleOpts {
+            id,
+            target: Vec::new(),
+            lossy: false,
+            flush_timeout: None,
+            key: None,
+            cert_pem: None,
+            key_pem: None,
+            root_cert_pem: None,
+            control: None,
+            read_only: false,
+        },
+    }
+}
+
+// Test function that expects a very specific type of volume and expects
+// the generation number provided to match the generation number in the
+// sub-volume.  The value 0 in the expected_gen vec tells this test function
+// that index in the list of sub_volumes should not be a Region, and therefore
+// will not have a generation number field.
```

Comment:
Nit: We should use an `Option` for this; no need to overload the value of the generation number.